src/components/dialogs/create-contingency-list-dialog.js

```diff
@@ -187,9 +218,41 @@ export const CreateContingencyListDialog = ({ open, onClose }) => {
             handleCreateNewContingencyList();
         }
     };
+
+    const handlePopupConfirmation = () => {
+        setOpenConfirmationPopup(false);
+        setContingencyListType(chosenContingencyListType);
+        setCreateContingencyListErr('');
+        setUnsavedChanges(false);
+    };
+
+    const onScriptChangeHandler = (newScript) => {
+        setCurrentScript(newScript);
+        if (newScript !== currentScript) {
+            setUnsavedChanges(true);
+        }
+    };
+
+    const handleCriteriaBasedFilterCreation = (filter) => {
+        currentCriteriaBasedFilter.current = {};
```

Comment:
currentCriteriaBasedFilter.current = filter.equipmentFilterForm ?

Reply:
Why outdated ? You gonna change it ?

Reply:
Yes, the useRef was replaced with a useState.